tests/shippers/test_logstash.py

```diff
@@ -0,0 +1,193 @@
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+import datetime
+import gzip
+import http
+import http.server
+import os
+import ssl
+import threading
+from base64 import b64encode
+from typing import Any
+from unittest import TestCase
+from unittest.mock import MagicMock
+
+import pytest
+import responses
+import ujson
+from requests import PreparedRequest
+
+from shippers.logstash import _EVENT_SENT, _MAX_RETRIES, LogstashShipper
+
+_now = datetime.datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%S.%fZ")
+
+_dummy_event: dict[str, Any] = {
```

Comment:
not sure. everywhere we use `_dummy_event` without doing a `deepcopy` we are actually mutating it

Reply:
`_dummy_event` is only read, never modified, in these tests, unless you are afraid it can be modified by another test file

Reply:
I remember having a case similar to [this](https://github.com/elastic/elastic-serverless-forwarder/pull/210/files#diff-cc6c179c69e1fc063e981ba448ac2f6c1afc88e270d5e275d59db1966cce98feR78-R79)
where assigning `_dummy_event` to a new variable without `deepcopy` and mutating the new variable mutated also `_dummy_event`

```
PyDev console: starting.
Python 3.9.16 (main, Dec  7 2022, 10:16:11) 
[Clang 14.0.0 (clang-1400.0.29.202)] on darwin
_dummy_event = {"this": "that"}
my_event = _dummy_event
my_event["that"] = "this"
print(my_event)
{'this': 'that', 'that': 'this'}
print(_dummy_event)
{'this': 'that', 'that': 'this'}
```